### PR TITLE
LPD-26730 Do not update completed order payment status on exception

### PR DESCRIPTION
**Pull request by brianikim**
https://liferay.atlassian.net/browse/LPD-26730

@luca-pellizzon ,

Solution as we discussed. Implemented integration test

Sincerely,
Brian I. Kim

### CODE REVIEW

**Comment by brianikim** [comment]
ci:test:sf

**Review by luca-pellizzon — APPROVED** [review]

**Review by luca-pellizzon — COMMENTED** [review]
@brianikim for testing you could try using an offline payment method like "money order"

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:stop

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-payment-test/src/testIntegration/java/com/liferay/commerce/payment/internal/servlet/test/CommercePaymentServletTest.java [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+			commerceOrder.getCommerceOrderId(), cpInstance.getCPInstanceId(),
+			BigDecimal.ONE);
+
+		CommerceOrder checkoutOrder =
+			_commerceOrderEngine.checkoutCommerceOrder(
+				commerceOrder, _user.getUserId());
+
+		_mockHttpServletRequest = new MockHttpServletRequest("GET", "");
+
+		_mockHttpServletRequest.setAttribute("LOCALE", LocaleUtil.ITALY);
```
Comment (luca-pellizzon) [inline modules/apps/commerce/commerce-payment-test/src/testIntegration/java/com/liferay/commerce/payment/internal/servlet/test/CommercePaymentServletTest.java:206]: Nice :smile:

**Comment by luca-pellizzon** [comment]
@brianikim 
Portal log assert failure, see above log for more information:  User 45915 must have VIEW permission for com.liferay.commerce.model.CommerceOrder 45889 com.liferay.portal.kernel.security.auth.PrincipalException$MustHavePermission: User 45915 must have VIEW permission for com.liferay.commerce.model.CommerceOrder 45889  at com.liferay.portal.kernel.security.permission.resource.ModelResourcePermissionFactory$DefaultModelResourcePermission.check(ModelResourcePermissionFactory.java:101)  at com.liferay.portal.kernel.security.permission.resource.ModelResourcePermissionFactory$DefaultModelResourcePermission.check(ModelResourcePermissionFactory.java:77)  at com.liferay.portal.kernel.security.permission.resource.BaseModelResourcePermissionWrapper.check(BaseModelResourcePermissionWrapper.java:41)  at com.liferay.commerce.service.impl.CommerceOrderServiceImpl.getCommerceOrderByUuidAndGroupId(CommerceOrderServiceImpl.java:289)  at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)  at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)  at java.lang.reflect.Method.invoke(Method.java:498)  at com.liferay.portal.spring.aop.AopMethodInvocationImpl.proceed(AopMethodInvocationImpl.java:41)  at com.liferay.portal.spring.transaction.TransactionInterceptor.invoke(TransactionInterceptor.java:60)  at com.liferay.portal.spring.aop.AopMethodInvocationImpl.proceed(AopMethodInvocationImpl.java:48)  at com.liferay.portal.kernel.aop.ChainableMethodAdvice.invoke(ChainableMethodAdvice.java:55)  at com.liferay.portal.spring.aop.AopMethodInvocationImpl.proceed(AopMethodInvocationImpl.java:48)  at com.liferay.change.tracking.internal.aop.CTTransactionAdvice.invoke(CTTransactionAdvice.java:70)  at com.liferay.portal.spring.aop.AopMethodInvocationImpl.proceed(AopMethodInvocationImpl.java:48)  at com.liferay.portal.spring.aop.AopInvocationHandler.invoke(AopInvocationHandler.java:40)  at com.sun.proxy.$Proxy1308.getCommerceOrderByUuidAndGroupId(Unknown Source)  at com.liferay.commerce.payment.internal.util.CommercePaymentHttpHelperImpl.getCommerceOrder(CommercePaymentHttpHelperImpl.java:80)  at com.liferay.commerce.payment.internal.servlet.CommercePaymentServlet.doGet(CommercePaymentServlet.java:99)  at javax.servlet.http.HttpServlet.service(HttpServlet.java:529)  at javax.servlet.http.HttpServlet.service(HttpServlet.java:623)  at com.liferay.commerce.payment.internal.servlet.test.CommercePaymentServletTest.testCompletePaymentPersists(CommercePaymentServletTest.java:249)

There must be something wrong with permission management. We know that the payment status must stick to complete but it's also true that this exception must be managed somehow. We can even set the test to expect the exception.

**Comment by luca-pellizzon** [comment]
ci:forward:force

**Pull request closed without merging**